lib/tapioca/dsl/compilers/active_model_attributes.rb

```diff
@@ -99,8 +99,17 @@ def handle_method_pattern?(pattern)
           HANDLED_METHOD_TARGETS.include?(target.to_s)
         end
 
+        sig { params(attribute_name: ::String, attribute_type_value: T.untyped).returns(::String) }
+        def type_for(attribute_name, attribute_type_value)
+          type = if constant.respond_to?(:__tapioca_attribute_types)
```

Comment:
This change is first of its kind so we'd like to exhaust other options first.

Reply:
Totally understandable. Which other options are worth considering?

Reply:
I think eventually, no matter what, we'll need to be able to override the types that Tapioca is generating via compilers.

I'm dealing with another situation right now where the column type is a string, but a gem overrides the reader to return an object.

Reply:
Like I said in the description, this is meant to serve as an integration point for libraries. I don't think we'll have much luck guessing the type.

So for example, this interface would allow me to implement something like this in user land:
```ruby
attribute :user, sig: -> { User }
```

I don't think Tapioca should get into the runtime business.